pkg/aks/create.go

```diff
@@ -38,98 +39,123 @@ func CreateCluster(ctx context.Context, cred *Credentials, clusterClient service
 }
 
 // createManagedCluster creates a new managed Kubernetes cluster.
-func createManagedCluster(ctx context.Context, cred *Credentials, workplacesClient services.WorkplacesClientInterface, spec *aksv1.AKSClusterConfigSpec, phase string) (containerservice.ManagedCluster, error) {
-	managedCluster := containerservice.ManagedCluster{
-		ManagedClusterProperties: &containerservice.ManagedClusterProperties{},
+func createManagedCluster(ctx context.Context, cred *Credentials, workplacesClient services.WorkplacesClientInterface, spec *aksv1.AKSClusterConfigSpec, phase string) (*containerservice.ManagedCluster, error) {
+	managedCluster := &containerservice.ManagedCluster{
+		Name:     to.StringPtr(spec.ClusterName),
+		Location: to.StringPtr(spec.ResourceLocation),
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			KubernetesVersion: spec.KubernetesVersion,
+		},
 	}
 
-	// Get tags from config spec
-	tags := make(map[string]*string)
+	managedCluster.Tags = make(map[string]*string)
 	for key, val := range spec.Tags {
 		if val != "" {
-			tags[key] = to.StringPtr(val)
+			managedCluster.Tags[key] = to.StringPtr(val)
 		}
 	}
 
-	// Get network profile from config spec
-	var vmNetSubnetID *string
 	networkProfile := &containerservice.NetworkProfile{
-		NetworkPlugin:   containerservice.Kubenet,
-		NetworkPolicy:   containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)),
-		LoadBalancerSku: containerservice.Standard,
+		NetworkPolicy: containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)),
 	}
 
-	if spec.LoadBalancerSKU != nil {
-		networkProfile.LoadBalancerSku = containerservice.LoadBalancerSku(to.String(spec.LoadBalancerSKU))
+	switch containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)) {
+	case containerservice.NetworkPolicyCalico:
+		if containerservice.NetworkPlugin(to.String(spec.NetworkPlugin)) != containerservice.Kubenet {
+			return nil, fmt.Errorf("networkPolicy 'calico' is only supported with networkPlugin 'kubenet'")
+		}
 	}
 
-	if containerservice.NetworkPlugin(to.String(spec.NetworkPlugin)) == containerservice.Azure {
-		networkProfile.NetworkPlugin = containerservice.NetworkPlugin(to.String(spec.NetworkPlugin))
-		virtualNetworkResourceGroup := spec.ResourceGroup
+	switch to.String(spec.NetworkPlugin) {
+	case string(containerservice.Azure):
+		networkProfile.NetworkPlugin = containerservice.Azure
+	case string(containerservice.Kubenet):
+		networkProfile.NetworkPlugin = containerservice.Kubenet
+	case "":
+		networkProfile.NetworkPlugin = containerservice.Kubenet
+	default:
+		return nil, fmt.Errorf("networkPlugin '%s' is not supported", to.String(spec.NetworkPlugin))
+	}
+
+	switch to.String(spec.LoadBalancerSKU) { // TODO: only standard is supported for now, find a way to validate this
+	case string(containerservice.Standard):
+		networkProfile.LoadBalancerSku = containerservice.Standard
+	case string(containerservice.Basic):
+		logrus.Warnf("loadBalancerSKU 'basic' is not supported")
```

Comment:
I will test it if it is possible to create AKS with `loadBalancerSKU = containerservice.Basic`

Reply:
The problem is that even if it's not possible we already allow this option in API, if we remove this now it might be a breaking change for users. As we know from past experience some undocumented options can pass through azure API.